imops/interp2d.py

```diff
@@ -0,0 +1,37 @@
+from typing import Optional
+
+import numpy as np
+from scipy.spatial import KDTree
+
+from cpp_modules import Linear2DInterpolatorCpp
+
+
+class Linear2DInterpolator(Linear2DInterpolatorCpp):
+    """
+    2D delaunay triangulation and parallel linear interpolation
+    Example:
+    n, m = 1024, 2
+    points = np.random.randint(low=0, high=1024, size=(n, m))
+    points = np.unique(points, axis=0)
+    x_points = points[: n // 2]
+    values = np.random.uniform(low=0.0, high=1.0, size=(len(x_points),))
+    interp_points = points[n // 2:]
+    n_jobs = -1 # will be equal to num of CPU cores
+    interpolator = Linear2DInterpolator(x_points, values, n_jobs)
+    # Also you can pass values to __call__ and rewrite the ones that were passed to __init__
+    interp_values = interpolator(interp_points, values + 1.0, fill_value=0.0)
+    """
+
+    def __init__(self, points: np.ndarray, values: np.ndarray, n_jobs: int = 1, **kwargs):
```

Comment:
Let's rename `n_jobs` by `num_threads` to maintain consistency with the rest of `imops`. Also let's make it argument of `__call__`